RackApp/app.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require 'erb'
+require 'rack'
+require 'active_support/time'
+
+class MyController
+  attr_reader :person, :time
+
+  def initialize(user, time)
+    @person = user
+    @time = time
+  end
+
+  def file(format)
+    File.read(File.join(__dir__, "template.#{format}.erb"))
+  end
+
+  def template(format)
+    ERB.new(file(format))
+  end
+
+  def render(format)
+    template(format).result(binding)
+  end
+end
+
+class User
+  attr_reader :nickname
+
+  def initialize(nickname)
+    @nickname = nickname
+  end
+end
+
+class CountryTime
+  attr_reader :zone, :timezone
```

Comment:
`attr_reader` method creates a getter method within `CountryTime` class, getter class means/looks like this:

```
def timezone
  @timezone 
end
```

so by defining `def timezone` again, you're rewriting the created getter created by `attr_reader :timezone`

so either you use the created getter or create your own:

1. created getter by `attr_reader :timezone`

```
class CountryTime

  attr_reader :zone, :timezone

  def initialize(zone)
    @zone = zone
    Time.zone = zone
    @timezone = Time.zone.now
  end
end
```

2. manually created "getter" by a method

```
class CountryTime
  attr_reader :zone

  def initialize(zone)
    @zone = zone
  end

  def timezone
    Time.zone = zone
    Time.zone.now
  end
end
```

I would prefer the first way in this case, so `timezone` is being generated only once, it the second case you would generate the timezone in every method call.